### PR TITLE
ci(deps): add more dependencies

### PR DESCRIPTION
**Pull request by Track07-cda**
Add some useful dependencies
Fix tensorflowjs no be able to install

### CODE REVIEW

**Review by ShaneHe023 — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -8,8 +8,15 @@ authors = [
 dependencies = [
     "phiflow>=2.4.0",
     "tensorflow>=2.13.0",
-    "tensorflowjs>=4.10.0",
+    "tensorflowjs",
```
Comment (Lutra-Fs) [inline pyproject.toml:11]: Needs further digging on the tfjs converter, I am aware of the capability with browser

**Review by Lutra-Fs — CHANGES_REQUESTED** [review]
Will do further check tmr, better hold this PR now

File: pyproject.toml [hunk 2]
```diff
@@ -8,8 +8,15 @@ authors = [
...
     "onnx>=1.14.0",
+    "dill",
+    "numpy<1.24",
+    "tqdm",
+    "matplotlib",
+    "ipykernel",
+    "jupyter",
+    "jupyterlab"
```
Comment (Lutra-Fs) [inline pyproject.toml]:
```suggestion
    "tqdm>=4.65.0",
    "matplotlib>=3.7.2",
    "ipykernel>=6.25.0",
    "jupyter>=1.0.0",
    "jupyterlab>=4.0.4",
    "dill>=0.3.7",
    "tensorflow-io-gcs-filesystem==0.31.0",
    "tf2onnx>=1.8.4",
    "onnxruntime>=1.15.1",
    "numpy>=1.24.3",
```

**Review by Lutra-Fs — CHANGES_REQUESTED** [review]
Add version constraints to reduce the installation time. Also pls change the PR title following commit message convention

**Review by Lutra-Fs — APPROVED** [review]

**Pull request merged**